.github/workflows/ci.yml

```diff
@@ -233,13 +247,32 @@ jobs:
         # Allows installing for a different base target
         target: ${{ matrix.rust.target || matrix.target }}
 
+    - name: Set up Cargo Dinghy
+      if: matrix.dinghy
+      run: |
+        cargo install cargo-dinghy --target x86_64-apple-darwin
+
+        # Launch the simulator
+        xcrun simctl list runtimes
+        RUNTIME_ID=$(xcrun simctl list runtimes | grep iOS | cut -d ' ' -f 7 | tail -1)
+        export SIM_ID=$(xcrun simctl create My-iphone7 com.apple.CoreSimulator.SimDeviceType.iPhone-7 $RUNTIME_ID)
+        xcrun simctl boot $SIM_ID
+
+        # Overwrite `cargo` to point to `cargo-dinghy`
+        mkdir bin
+        echo "#!/bin/bash" > bin/cargo
+        echo "cargo-dinghy --device=$SIM_ID \"\$@\"" >> bin/cargo
```

Comment:
I see why you did this but think it may save you future time to just have the `cargo dinghy test` section in this run-block and below in the `Test without features` just skip it it with an `if: ${{ !matrix.dinghy }}`. Overriding the  `cargo` with a new shell-script cargo feels a bit fragile (maybe the `actions-rs/cargo` action changes to not look at `PATH` for it's cargo) and isn't something a lot of people would expect.

Reply:
Yeah you're right, I did it like this because I was hoping the other scripts would work as well, but it turns out `cargo dinghy` doesn't support specifying features from a workspace root, so it's kind of useless like this anyhow.

Reply:
Oh, that makes more sense. I was thinking you did it for ease of use with the `actions-rs/cargo` action.

> it turns out cargo dinghy doesn't support specifying features from a workspace root, so it's kind of useless like this anyhow.

Yeah. Dinghy isn't perfect. When I added [fixed iOS simulator support](https://github.com/sonos/dinghy/pull/96), I wasn't in love with how I did it but it's better than nothing. Looking back, I'm glad I added it as I see more and more iOS CI.